### PR TITLE
[Merged by Bors] - node: crash if specified config can't be loaded

### LINKED ISSUE #4160: Node should crash when the specified config is invalid
When the `go-spacemesh` receives wrong config it logs
``` failed loading config from /home/XXX/services/node_config.json trying ./config.toml. error While parsing config: invalid character ']' looking for beginning of value```

And then it basically loads defaults. 

I think this behavior is really dangerous. The wrong config should result with a failed start.

### PR DESCRIPTION
**Pull request by pigmej**
Closes #4160

changes:

* if config file is specified we don't fallback to anything and instead load that config file
* otherwise load config based on the specified preset or regular default config

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -195,6 +196,10 @@ func LoadConfig(fileLocation string, vip *viper.Viper) (err error) {
 	err = vip.ReadInConfig()
```
Comment (countvonzero) [inline config/config.go]: just do `return vip.ReadInConfig()` here. the upstream caller does log.fatal() on error already.

in essence, we simply don't fall back to do default config.

Comment (pigmej) [reply config/config.go]: @countvonzero yeah, but the problem is that this logic is needed in tests, and we pretty much somehow assume kinda everywhere that this is tried to be ALWAYS read.

That logic is fatally flawed and should be reworked completely.

Nevertheless, that fix is meant to fix the problem faced in the real production application now.

Comment (dshulyak) [reply config/config.go]: i refactored to return fatal error on any failure to load specified config file.
if config file is not specified we will try to load default config file if it exists

**Review by dshulyak — APPROVED** [review]

**Comment by dshulyak** [comment]
@countvonzero @pigmej please check if what i wrote in description sounds good, i simplified config handling a bit

**Comment by dshulyak** [comment]
bors try

**Comment by pigmej** [comment]
bors try

File: config/config.go [hunk 2]
```diff
@@ -186,26 +185,20 @@ func defaultTestConfig() BaseConfig {
...
-			log.Warning("failed loading config from %v trying %v. error %v", fileLocation, defaultConfigFileName, err)
-			vip.SetConfigFile(defaultConfigFileName)
-			err = vip.ReadInConfig()
-		}
-		// we change err so check again
-		if err != nil {
-			return fmt.Errorf("failed to read config file %v", err)
+func LoadConfig(config string, vip *viper.Viper) error {
+	if config == defaultConfigFileName || config == "" {
+		if _, err := os.Stat(defaultConfigFileName); err != nil {
```
Comment (countvonzero) [inline config/config.go]: i honestly don't understand the purpose of `defaultConfigFileName`, which is set to `config.toml`. this file was deleted a while back i think.

the code would be simpler without this variable, and move on (return nil) `config == ""`

**Review by countvonzero — APPROVED** [review]

File: config/config.go (hunk 2 above)
Comment (dshulyak) [reply config/config.go]: i thought that it is kind of a ux thing, but you are right there is no need for it

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Review by fasmat — APPROVED** [review]

**Pull request closed without merging**